data-service/src/main/java/com/amazonaws/blox/dataservice/api/StartDeploymentApi.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.api;
+
+import com.amazonaws.blox.dataservice.mapper.ApiModelMapper;
+import com.amazonaws.blox.dataservice.model.Environment;
+import com.amazonaws.blox.dataservice.model.EnvironmentId;
+import com.amazonaws.blox.dataservice.model.EnvironmentRevision;
+import com.amazonaws.blox.dataservice.repository.EnvironmentRepository;
+import com.amazonaws.blox.dataservicemodel.v1.exception.InternalServiceException;
+import com.amazonaws.blox.dataservicemodel.v1.exception.ResourceNotFoundException;
+import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.StartDeploymentRequest;
+import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.StartDeploymentResponse;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Slf4j
+@Component
+@AllArgsConstructor
+public class StartDeploymentApi {
+
+  @NonNull private final ApiModelMapper apiModelMapper;
+  @NonNull private final EnvironmentRepository environmentRepository;
+
+  public StartDeploymentResponse startDeployment(final StartDeploymentRequest request)
+      throws ResourceNotFoundException, InternalServiceException {
+    try {
+      final EnvironmentId environmentId =
+          apiModelMapper.toModelEnvironmentId(request.getEnvironmentId());
+      final Environment environment = environmentRepository.getEnvironment(environmentId);
+
+      final EnvironmentRevision environmentRevision =
+          environmentRepository.getEnvironmentRevision(
+              environmentId, request.getEnvironmentRevisionId());
+
+      environment.setActiveEnvironmentRevisionId(environmentRevision.getEnvironmentRevisionId());
+
```

Comment:
Is the "lastUpdatedTime" in Environment needed to be updated?

Reply:
It should be updated, but I would like this update will be done on DB API level, which the `EnvironmentRepositoryDDB#updateEnvironment' will update this field.

I will do other PR for the missing part of the updateEnvironment method.

Reply:
I kind of lean to have all of the fields that need to be updated be set in the data-service api level, and leave the DB layer only take care of interacting with DB. But open to discussion.

Reply:
I think any call to make a "updateEnvironment" DDB call should update the `lastUpdatedTime`. In this case, we don't have to duplicate the `setLastUpdatedTime(...)`  before every `updateEnvironment` call.   Unless we have different propose of the latestUpdatedTime with the record, it's more like a way to simplify our code.